### PR TITLE
Enable conditional attributes

### PR DESCRIPTION
**Pull request by Jasopaum**

### CODE REVIEW

File: client/src/components/mapping/TabColumnPicking/DynamicColumnPicker/index.tsx [hunk 1]
```diff
@@ -35,31 +38,42 @@ const DynamicColumnPicker = ({ attribute, schema, source }: Props) => {
...
+  );
   const attributesForResource = useSelector(
     (state: IReduxStore) => state.resourceInputs.attributesMap
   );
+
+  const path = attribute.path;
   let attributeId = attributesForResource[path]
     ? attributesForResource[path].id
     : null;
+  let inputGroupId =
```
Comment (simonvadee) [inline client/src/components/mapping/TabColumnPicking/DynamicColumnPicker/index.tsx:52]: please explain wtf is going on here, this is too hard to read without a pint of coffee

File: client/src/components/mapping/TabColumnPicking/StaticValueForm/index.tsx [hunk 2]
```diff
@@ -62,6 +68,12 @@ const StaticValueForm = ({ attribute }: Props): React.ReactElement => {
   let attributeId = attributesForResource[path]
     ? attributesForResource[path].id
     : null;
+  let inputGroupId =
```
Comment (simonvadee) [inline client/src/components/mapping/TabColumnPicking/StaticValueForm/index.tsx:71]: same

File: client/src/graphql/mutations/addConditionToInputGroup.graphql [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#import "../fragments/entireCondition.graphql"
+
+mutation addConditionToInputGroup(
+  $inputGroupId: ID!
+  $action: String
```
Comment (simonvadee) [inline client/src/graphql/mutations/addConditionToInputGroup.graphql:5]: isn't this required ? as well as the other fields ?

File: client/src/services/selectedNode/reducer.tsx [hunk 4]
```diff
@@ -3,7 +3,8 @@ import { ISimpleAction } from 'types';
 const initialState: any = {
   source: null,
   resource: null,
-  attribute: null
+  attribute: null,
+  selectedInputGroup: 0
```
Comment (simonvadee) [inline client/src/services/selectedNode/reducer.tsx:7]: i don't think it's a good idea to initialize this to 0, I would use null instead. Would that cause an issue elsewhere ?

File: client/src/graphql/mutations/addConditionToInputGroup.graphql (hunk 3 above)
Comment (Jasopaum) [reply client/src/graphql/mutations/addConditionToInputGroup.graphql:5]: I first create empty conditions and fill them after. So we could even remove these arguments for the addConditionToInputGroup mutation. What do you think?

File: client/src/components/mapping/InputCondition/index.tsx [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+      >
+        <Tag minimal={true}>VALUE</Tag>
+        <Tag intent={'primary'} large={true}>
+          <input
+            className="text-input"
+            value={value}
+            type="text"
+            placeholder="value..."
+            onChange={(e: React.ChangeEvent<HTMLInputElement>): void => {
+              // NOTE value updated in DB at each character. Put a submit button?
```
[the remark this answers is not in the document]
Comment (Jasopaum) [reply client/src/components/mapping/InputCondition/index.tsx:206]: I just made an issue for that (https://github.com/arkhn/pyrog/issues/347)

File: server/prisma/migrations/migrate.lock [hunk 6]
```diff
@@ -1,4 +1,5 @@
 # Prisma Migrate lockfile v1
 
 20200623144855-database-schema
-20200803162228-input-groups
\ No newline at end of file
+20200803162228-input-groups
```
Comment (simonvadee) [inline server/prisma/migrations/migrate.lock:4]: duplicate

**Pull request merged**